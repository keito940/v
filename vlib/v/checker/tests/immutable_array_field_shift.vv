struct A {
mut:
	i []int
}

struct B {
	a A
}

fn main() {
	mut b := B{
		a: A{}
	}
	b.a.i << 3
}
