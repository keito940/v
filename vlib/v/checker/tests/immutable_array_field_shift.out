vlib/v/checker/tests/immutable_array_field_shift.v:14:4: error: field `a` of struct `B` is immutable
   12|         a: A{}
   13|     }
   14|     b.a.i << 3
             ^
   15| }
